src/main/java/mate/jdbc/dao/CarDaoImpl.java

```diff
@@ -0,0 +1,223 @@
+                + "WHERE driver_id = ? AND cars.is_deleted = FALSE;";
+        List<Car> cars = new ArrayList<>();
+        try (Connection connection = ConnectionUtil.getConnection();
+                PreparedStatement getAllByDriverStatement = connection.prepareStatement(query)) {
+            getAllByDriverStatement.setLong(1, driverId);
+            ResultSet resultSet = getAllByDriverStatement.executeQuery();
+            Car car = null;
+            while (resultSet.next()) {
+                cars.add(getCar(resultSet));
+            }
+        } catch (SQLException e) {
+            throw new DataProcessingException("Couldn't delete car with id " + driverId, e);
+        }
+        for (Car car : cars) {
+            car.setDrivers(getDrivers(car.getId()));
+        }
+        return cars;
+    }
+
+    private Car getCar(ResultSet resultSet) throws SQLException {
+        Long id = resultSet.getObject("id", Long.class);
+        String model = resultSet.getString("model");
+        Car car = new Car();
+        car.setId(id);
+        car.setModel(model);
+        Long manufacturerId = resultSet.getObject("manufacturer_id", Long.class);
+        String manufacturerName = resultSet.getString("name");
+        String manufacturerCountry = resultSet.getString("country");
+        Manufacturer manufacturer = new Manufacturer();
+        manufacturer.setId(manufacturerId);
+        manufacturer.setName(manufacturerName);
+        manufacturer.setCountry(manufacturerCountry);
+        car.setManufacturer(manufacturer);
+        return car;
+    }
+
+    private List<Driver> getDrivers(Long carId) {
+        List<Driver> drivers = new ArrayList<>();
+        String query = "SELECT drivers.* "
+                + "FROM cars_drivers "
+                + "INNER JOIN drivers "
+                + "ON cars_drivers.driver_id = drivers.id "
+                + "WHERE car_id = ? AND is_deleted = FALSE;";
+        try (Connection connection = ConnectionUtil.getConnection();
+                PreparedStatement getDriversStatement
+                        = connection.prepareStatement(query)) {
+            getDriversStatement.setLong(1, carId);
+            ResultSet resultSet = getDriversStatement.executeQuery();
+            while (resultSet.next()) {
+                drivers.add(getDriverFromResultSet(resultSet));
+            }
+            return drivers;
+        } catch (SQLException e) {
+            throw new DataProcessingException("Can`t get drivers for carId: " + carId, e);
+        }
+    }
+
+    private Driver getDriverFromResultSet(ResultSet resultSet) throws SQLException {
+        Long driverId = resultSet.getObject("id", Long.class);
+        String driverName = resultSet.getString("name");
+        String driverLicense = resultSet.getString("license_number");
+        Driver driver = new Driver();
+        driver.setId(driverId);
+        driver.setName(driverName);
+        driver.setLicenseNumber(driverLicense);
```

Comment:
we have a constructor for Driver(name, licenseNumber)

Reply:
In this dao, I used the strategy of creating an object through setters)